agent/src/flow_generator/protocol_logs/mq/rocketmq.rs

```diff
@@ -97,9 +98,63 @@ pub struct RocketmqInfo {
     endpoint: Option<String>,
 }
 
+// 添加解析 (UNIQ_)KEY , traceparent 和 sw8 的辅助函数 (从 properties 字符串中)
+fn parse_trace_info_from_properties(
+    properties: &str,
+) -> (Option<String>, Option<String>, Option<String>) {
+    let mut msg_key = None;
+    let mut trace_id = None;
+    let mut span_id = None;
+
+    // 使用 STX 控制字符 (U+0002) 作为分隔符分割不同的属性对
+    let pairs: Vec<&str> = properties.split(|c| c == '\u{2}').collect();
+
+    for pair in pairs {
+        if pair.is_empty() {
+            continue;
+        }
+
+        // 使用 SOH 控制字符 (U+0001) 作为分隔符分割每个属性的键和值
+        let kv: Vec<&str> = pair.splitn(2, '\u{1}').collect();
+        if kv.len() == 2 {
+            let key = kv[0];
+            let value = kv[1];
+
```

Comment:
```suggestion
        let mut iter = pair.splitn(2, '\u{1}');
        if let (Some(key), Some(value)) = (iter.next(), iter.next()) {
```
rust 保证执行顺序
https://doc.rust-lang.org/stable/reference/expressions.html?highlight=left#evaluation-order-of-operands